Allow Entity List Passed Through

toriannwin opened the pull request:
## Hey, I just made a Pull Request!

Related to issue: [Bazaar: Entity List Not Passed Through React Components](https://github.com/backstage/community-plugins/issues/3881)

With this change, a list of catalogEntities passed into `ProjectPreview` will pass all the way through to the `HomePageBazaarInfoCard`, which means that if a custom list of components was originally passed in, only those list of components (unlinked) will appear as link options.

Consider a case where the application should only allow Components, not Resources, to be assigned as the Entity for a Bazaar project:

Old Behavior / using SortView without passing in the optional allCatalogEntities:
<img width="667" alt="image" src="https://github.com/user-attachments/assets/60569c37-ee24-4657-9237-70bd931d2232" />

New Behavior / passing allCatalogEntities into SortView:

<img width="670" alt="image" src="https://github.com/user-attachments/assets/18316017-f4e6-4ac9-a8b6-e0aba5e4efc3" />

**To make this more robust, I've allowed either a list of `catalogEntities` or a `filter` `EntityFilterQuery` object to be passed in through these components. As the props documentation says, catalogEntities will always take priority.** For examples:
1. If a user passes in a list of catalogEntities to display as the entity link options, those are what will display.
2. If they omit catalogEntities and pass in a custom `filter`, then `fetchCatalogItems` will use the passed in filter. 
3. If both are omitted, `fetchCatalogItems` will proceed as previously written with defaulting to Component and Resource kinds.
4. If they pass in both, `catalogEntities` will take priority.


#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/community-plugins/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/community-plugins/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

vinzscam commented on workspaces/bazaar/plugins/bazaar/src/components/HomePageBazaarInfoCard/HomePageBazaarInfoCard.tsx:
```diff
@@ -93,7 +95,12 @@ export const HomePageBazaarInfoCard = ({
 
   const [catalogEntities, fetchCatalogEntities] = useAsyncFn(async () => {
-    const entities = await fetchCatalogItems(catalogApi);
+    let entities = [];
```
do you think it could make sense to remove the fetch completely or it makes sense to keep it here for some cases?

toriannwin replied:
and yeah same here, was thinking backwards compatibility

vinzscam reviewed (changes requested):
🙏

toriannwin commented on workspaces/bazaar/plugins/bazaar/src/components/SortView/SortView.tsx:
```diff
@@ -97,6 +102,7 @@ export const SortView = (props: SortViewProps) => {
     useState<Entity[]>();
 
   const [catalogEntities, fetchCatalogEntities] = useAsyncFn(async () => {
+    if (allCatalogEntities) return allCatalogEntities;
```
I am looking back at this now though based on your feedback on the issue to see about just passing through a filter value rather than having to pre-filter and pass in catalog entities

toriannwin commented:
@vinzscam apologies for the delay, but I've made some changes to allow an optional filter to be passed through the components as well, and added both this and the catalogEntities functionality to be passed into the BazaarOverviewCard as well!

As I added in the PR description:

I've allowed either a list of catalogEntities or a filter EntityFilterQuery object to be passed in through these components. As the props documentation says, catalogEntities will always take priority. For examples:

1. If a user passes in a list of catalogEntities to display as the entity link options, those are what will display.
2. If they omit catalogEntities and pass in a custom filter, then fetchCatalogItems will use the passed in filter.
3. If both are omitted, fetchCatalogItems will proceed as previously written with defaulting to Component and Resource kinds.
4. If they pass in both, catalogEntities will take priority.

04kash commented on workspaces/bazaar/.changeset/cuddly-beans-exercise.md:
```diff
@@ -0,0 +1,7 @@
+'@backstage-community/plugin-bazaar': patch
+---
+
+Catalog Entities can be passed all the way through to the HomePageBazaarInfoCard so custom entities are maintained when opening the Link Entity menu
+OR
+A custom filter can be passed in to bazaar components to override the default filter in fetchCatalogItems.
```
```suggestion
Enables passing catalog entities through to `HomePageBazaarInfoCard`, ensuring custom entities are maintained when using the Link Entity menu.
Also adds support for providing a custom filter to Bazaar components to override the default filter in `fetchCatalogItems`.
```

04kash reviewed (changes requested):
Thanks @toriannwin, could you also mention how to pass in `allCatalogEntities` and ` filter` to bazaar components in the README [here](https://github.com/backstage/community-plugins/blob/main/workspaces/bazaar/plugins/bazaar/README.md)?
